keras/layers/convolutional.py

```diff
@@ -321,13 +330,18 @@ def __init__(self, filters,
                  kernel_constraint=None,
                  bias_constraint=None,
                  **kwargs):
+        if padding == 'causal':
```

Comment:
was causal mode tested with channels last? I don't recall having facing any issues with data format when I opened PR for this

Reply:
This test case is failing.

Reply:
Please see comment above

Reply:
I tested this locally by pulling your changes. It appears that there's an issue with conv1d with all of the paddings when data format is channels_last. Conv1d works properly for channels_first, with any padding. As discussed, we should update the error message to indicate this issue with conv1d